ArcGISRuntimeSDKQt_QMLSamples/DisplayInformation/ReadSymbolsFromMobileStyle/ReadSymbolsFromMobileStyle.qml

```diff
@@ -0,0 +1,339 @@
+// [WriteFile Name=ReadSymbolsFromMobileStyle, Category=DisplayInformation]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.11
+import QtQuick.Controls 2.5
+import QtQuick.Layouts 1.3
+import Esri.ArcGISRuntime 100.6
+import Esri.ArcGISExtras 1.1
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    property url dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/styles/emoji-mobile.stylx"
+    property var currentSymbol
+    property var currentFace
+    property var currentEyes
+    property var currentMouth
+    property var currentHat
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Map {
+            BasemapTopographic {}
+        }
+
+        GraphicsOverlay {
+            id: graphicsOverlay
+        }
+
+        onMouseClicked: {
+            var graphic = ArcGISRuntimeEnvironment.createObject("Graphic");
+            graphic.geometry = mouse.mapPoint;
+            graphic.symbol = currentSymbol;
+            graphicsOverlay.graphics.append(graphic);
+        }
+
+        Button {
+            anchors {
+                bottom: mapView.attributionTop
+                horizontalCenter: parent.horizontalCenter
+                bottomMargin: 10
+            }
+            text: "Clear Graphics"
+            onClicked: graphicsOverlay.graphics.clear();
+        }
+    }
+
+    SymbolStyle {
+        id: symbolStyle
+        styleLocation: dataPath
+        Component.onCompleted: load()
+
+        onLoadStatusChanged: {
+            // fetch the default parameters
+            if (loadStatus === Enums.LoadStatusLoaded) {
+                defaultSearchParameters();
+            }
+        }
+
+        onDefaultSearchParametersStatusChanged: {
+            // search the symbol style with defaults - all symbols will be returned
+            if (defaultSearchParametersStatus === Enums.TaskStatusCompleted) {
+                searchSymbols(defaultSearchParametersResult);
+            }
+        }
+
+        onSearchSymbolsStatusChanged: {
+            if (searchSymbolsStatus !== Enums.TaskStatusCompleted)
+                return;
+
+            // get initial list of symbols
+            searchSymbolsResult.forEach(function(result){
+                if (result.category === "Face") {
+                    currentFace = result;
+                }
```

Comment:
Is there a chance you don't need to loop through everything here? Perhaps change to a traditional for-loop if you need to break early.